### DISCUSSION TITLE
Invalid payload

### DISCUSSION THREAD

**Discussion by DmitryPosnov** [socketio/socket.io#4976] [Q&A] [answered]
I get a parser error when more than 6 clients connect to the server. Always the same parser error.
Properties maxHttpBufferSize: 1e8, pingTimeout: 60000 installed as in the documentation.
![invalid payload](https://github.com/socketio/socket.io/assets/163400365/cc47fdee-4ec0-4993-bccd-8117426c7142)

JSON which I can see in the developer tab like this. 
![trunceted](https://github.com/socketio/socket.io/assets/163400365/38f64973-66ef-49e1-ae99-6c597c7d4ec4)

Here it breaks off in the middle of the message that I send from the server, but on the server side it sends complete message without any error.

i use
"socket.io": "^4.7.2",
"socket.io-client": "^4.7.2",

with
"socket.io": "^4.7.4",
"socket.io-client": "^4.7.4",
same error.

I tried other parsers (socket.io-json-parser, socket.io-msgpack-parser), but the error is the same.

**Comment by darrachequesne** [comment]
Hi! That's weird indeed. Could you please provide the payload that triggers the parser error?

**Comment by DmitryPosnov** [comment]
Hi! Yes!
My socket server config:
        {         
            path: process.env.SOCKET_PATH,
            maxHttpBufferSize: 1e8,
            pingTimeout: 60000,
            cors: {
                    origin: '*'
                },
            addTrailingSlash: false,  
            transports: ['polling'],
            connectionStateRecovery: {
                    maxDisconnectionDuration: 300000,
                    skipMiddlewares: true
                }
        }

Method for emit JSON from server:
socketOnGetSockets(socket) {
        socket.on(socketEvents.getSockets, async () => {
            const sockets = await this.getSockets(socket)   //// under the hood: (for...of) io.fetchSockets() for cteare my JSON
            console.log('--------------------------')
            console.log(sockets)
            console.log('--------------------------')
            socket.emit(socketEvents.sockets, sockets)
        });
    }

My JSON to send from the server to socket client:
'--------------------------'
{
  amount: 7,
  error: false,
  sockets: [
    {
      id: '92AytsDl4noA2q6LAAAN',
      user: 'Администратор',
      sessionID: '93025',                          // type string
      room: 'roomID:93025',
      timestamp: 2024-03-15T03:56:33.000Z
    },
    {
      id: 'm6ZSSp_BpH5eH55xAAAS',
      user: 'Администратор',
      sessionID: '10366',                           // type string
      room: 'roomID:10366',
      timestamp: 2024-03-15T03:56:34.000Z
    },
    {
      id: 'klMSIB6qilVXBUxPAAAU',
      user: 'Администратор',
      sessionID: '25269',                          // type string
      room: 'roomID:25269',
      timestamp: 2024-03-15T03:56:34.000Z
    },
    {
      id: 'c3T5eO7IO7rQG3mhAAAX',
      user: 'Администратор',
      sessionID: 62947,                          // !!! type number, whether it matters or not
      room: 'roomID:62947',
      timestamp: 2024-03-15T03:56:57.000Z
    },
    {
      id: 'OkGNaJTah3mBVF9YAAAa',
      user: 'Администратор',
      sessionID: 49853,                          // !!! type number, whether it matters or not
      room: 'roomID:49853',
      timestamp: 2024-03-15T03:57:02.000Z
    },
    {
      id: 'GS9pHEpU8DjV_fZ1AAAd',
      user: 'Администратор',
      sessionID: 18705,                           // !!! type number, whether it matters or not
      room: 'roomID:18705',
      timestamp: 2024-03-15T03:57:08.000Z
    },
    {
      id: 'IWQqWSyTHZjC59HsAAAg',
      user: 'Администратор',
      sessionID: 90050,                           // !!! type number, whether it matters or not
      room: 'roomID:90050',
      timestamp: 2024-03-15T03:57:22.000Z
    }
  ]
}
'--------------------------'

Browser console:
![15 03 br](https://github.com/socketio/socket.io/assets/163400365/29e5619f-6fbc-423d-8b8d-5db5f7c284e6)


Browser reseived JSON
![15 03 br json](https://github.com/socketio/socket.io/assets/163400365/ebe895a7-9304-469b-a7cf-17cf960804be)


After an invalid payload error, the client socket reconnected and the socket server created a new socket without restoring the old socket, meaning the old socket remained on the server. If the client requests the same JSON again then the invalid payload errors will occur again - the client socket has reconnected and the socket server has created a new socket without recovery and so on.
On my server there were up to 300 sockets on the server, although in reality 7 client browsers were open

**Comment by RonnieRocket147** [comment]
That last screenshot indicates to me that the JSON payload is truncated as the name `user` field is not complete and contains a partial UTF-8 character

**Comment by DmitryPosnov** [comment]
Hi! I changed the `user ` field to all users to simple "user".
if connected 7 clients then it is OK:
![image](https://github.com/socketio/socket.io/assets/163400365/555b4664-f924-43d6-b785-838694a9a41d)

I noticed that if the Content-type: Length is less than 1000 then everything is OK, but if it is more than 1000 then there is an error
![image](https://github.com/socketio/socket.io/assets/163400365/761446b5-7cde-4909-b1ea-3a7fde72f34c)


but if i connected 8 clients the error is the same: parse error Error: invalid payload
![image](https://github.com/socketio/socket.io/assets/163400365/421a5dce-1df6-4415-9076-5d9a6f1fc6cf)

Maybe I need to specify additional headers?

**Comment by DmitryPosnov** [comment]
I added for 'httpCompression' transport polling settings from https://socket.io/docs/v4/server-options/ and it works perfect
  httpCompression: {
    // Engine.IO options
    threshold: 2048, // defaults to 1024
    // Node.js zlib options
    chunkSize: 8 * 1024, // defaults to 16 * 1024
    windowBits: 14, // defaults to 15
    memLevel: 7, // defaults to 8
  }
i connected up to 13 clients, I didn't check further.

What settings should I use for compression then?

**Reply by darrachequesne** [reply, marked as answer]
Interesting! Thanks for digging into this. It seems something between your server and your client does not properly handle gzip compression. Do you have any proxy (nginx, httpd, ...) in-between?

As a temporary workaround, you can set `httpCompression` to `false`.

**Reply by DmitryPosnov** [reply]
I'm using the Next.js with built-in server without proxy.
I set httpCompression to false, and everything works better because if I left the settings for httpCompression, maximum only 13 clients can be connected without parser error.
Thanks for the help.

**Reply by darrachequesne** [reply]
We have added a guide to use Socket.IO with Next.js: https://socket.io/how-to/use-with-nextjs

However, I was not able to reproduce the compression error, were you able to find the culprit?

**Reply by DmitryPosnov** [reply]
I haven't been able to find the reason yet.
I'll try disabling compression on Nextjs' built-in server. Maybe it has some influence.
